tests/rptest/services/datalake/catalog/databricks_unity.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from typing import Dict, Optional
+
+from rptest.context.databricks import DatabricksContext
+from rptest.services.catalog_service import CatalogService, CatalogType
+from rptest.services.databricks_workspace import DatabricksCatalogInfo
+
+
+class DatabricksUnity(CatalogService):
+    def __init__(self, ctx, *, cloud_storage_bucket: str,
+                 catalog: DatabricksCatalogInfo):
+        super().__init__(
+            ctx,
+            cloud_storage_bucket,
+            num_nodes=0,
+        )
+
+        self._databricks_context = DatabricksContext.from_context(ctx)
+        self.warehouse_name = catalog.name
+        self.compute_warehouse_path()
+
+    def catalog_type(self) -> CatalogType:
+        return CatalogType.DATABRICKS_UNITY
+
+    @property
+    def iceberg_rest_url(self) -> str:
+        return self._databricks_context.iceberg_rest_url
+
+    def _configure_client(self, conf: Dict[str, Optional[str]]) -> None:
+        conf["token"] = self._databricks_context.token
+        # TODO: Understand why CatalogService sets warehouse to a cloud storage
+        # uri like s3://bucket/... and it works for all the tests but not
```

Comment:
Yea, it's quite messy... open to discussing more but I lean towards unifying REST catalogs to always supply the warehouse name, and have only REST_HDFS configure warehouse as the location.

Reply:
Odd about locations not working though. I wonder if internally DBX has a different mapping of warehouses to locations that the user-provided "locations". Does it work with PyIceberg?

Reply:
This is the config for pyiceberg.

Reply:
🤦

Ah yea interesting. My mental model (or at least, how I wish things were if they aren't) is that when using a REST catalog we _should_ always be passing the warehouse name rather than a warehouse location. Seems like the HadoopCatalog is the only catalog that really needs this to be a location